### ISSUE TITLE
Voided answer concepts showing up in App designer

### ISSUE THREAD

**Issue by mahalakshme** [avniproject/avni-webapp#1113]
### Describe the bug
Currently coded answer concepts which have been removed from the coded concept still show up in App Designer. 

### To Reproduce
Steps to reproduce the behaviour:

- Create a coded concept with A, B, C and D as answers to coded concept
- Now, remove A as the answer to the coded concept
- But when we search for concept A on App designer, It will still show as answer to the coded concept

Refer [this](https://github.com/avniproject/avni-webapp/assets/10369963/01875bbe-777c-43ff-b119-0e3bcc8b708e
) video to understand the issue: 

### Expected behavior
The removed concept A should not show up as part/answer to the coded concept

**Comment by ombhardwajj** [comment]
@mahalakshme , @vinayvenu I have fixed this issue and submitted a PR. Please have a look!

**Comment by petmongrels** [comment]
@ombhardwajj

### Some background.
This is the domain model of Concept answers.
- Concept entity has one to many ConceptAnswer entity
- ConceptAnswer entity has one to one to with child Concept entity

### What should happen in remove answer
- ConceptAnswer should be voided
- nothing should be done to child Concept (as it may be used elsewhere)

### Implementing this when the new Concept Answer has not been saved even once
- To identify this you will see that id/uuid will be empty/null
- Just remove it from the Concept Answer array

### Implementing this when the new Concept Answer has not been saved and being removed
- When it is removed void it in web app. Server call is not required yet.
- When save is pressed, web app should send the array consisting of voided ConceptAnswer also. The server should persist voided ConceptAnswers also by setting the field.

**Comment by mahalakshme** [comment]
@ombhardwajj are the above code review comments resolved?

**Comment by ombhardwajj** [comment]
@mahalakshme  No they have not been resolved , I am currently working on metabase issue and will only be able to work on this once that issue is complete!

**Comment by mahalakshme** [comment]
@ombhardwajj would you be able to address the above comments? Just wanted to know, so that if you don't have time, someone within the team can pick it up

**Comment by ombhardwajj** [comment]
@mahalakshme  I will address it. Please give me a week's time!

**Comment by ombhardwajj** [comment]
Hey @petmongrels @mahalakshme 
I've made a change in the backend in order to fix this bug. Have a look [here](https://github.com/avniproject/avni-server/commit/b2364e414f8da0e32a7731f2bc5407595a3e1df9)

I added a new method in `ConceptAnswerRepository`, `findByAnswerConceptAndIsVoidedFalse`, to retrieve only non voided ConceptAnswer entries associated with a given answerConcept.

I modified the `addDependentConcepts` method in `ConceptService` to use this new repository method, ensuring only non voided answers are included in the usage data returned by the /web/concept/usage/{uuid} endpoint.

I've tested it out and it works as expected. Let me know your thoughts on this approach.
